### PR TITLE
MF-647: Introduce PatientTable component

### PR DESCRIPTION
**Pull request by samuelmale**
An example usage of the PatientTable extension in a different MF:

### CODE REVIEW

**Comment by samuelmale** [comment]
@nmalyschkin I think we can later on introduce the 'advance filters' and 'column(s) controller' widgets.

File: packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import React, { useMemo, CSSProperties } from 'react';
+import styles from './patient-table.component.scss';
+
+const PatientTable: React.FC<PatientTableProps> = ({ patients, columns, search, pagination, isLoading, autoFocus }) => {
+  const rows: Array<any> = useMemo(
+    () =>
+      patients.map((patient) => {
+        const row = {};
+        columns.forEach((column) => {
+          row['id'] = patient['id'] || patient['uuid'];
```
Comment (nmalyschkin) [inline packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx:25]: This does not have to be done for every column.

File: packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+          const value = column.getValue ? column.getValue(patient) : patient[column.key];
```
Comment (nmalyschkin) [inline packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx:26]: `column.getValue?(patient) || patient[column.key]`

File: packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        });
+        return row;
+      }),
+    [patients, columns],
+  );
+  const handleSearch = React.useMemo(() => debounce((searchTerm) => search.onSearch(searchTerm), 300), []);
+
+  return (
+    <>
+      {!isLoading ? (
```
Comment (nmalyschkin) [inline packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx:47]: You could use this notation:
```tsx
if(isLoading)
  return <Skeleton />

return <Table />
```

Not a big difference, but make it easier to read

File: packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        <>
```
Comment (nmalyschkin) [inline packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx:48]: Right now the PatientTable returns two separate components, should be one.

**Review by nmalyschkin — COMMENTED** [review]
Great work! 👍

File: packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx (hunk 4 above)
Comment (samuelmale) [reply packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx:48]: Did you mean like the Skeleton and the Table should be under the same parent component :

```jsx

return (
<div> 
   {isLoading && <Skeleton />}
   {!isLoading && <Table/>}
</div>)

```

Or you suggest that it should have the Skeleton?

File: packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx (hunk 1 above)
Comment (samuelmale) [reply packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx:25]: Nice catch!

File: packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx (hunk 4 above)
Comment (FlorianRappl) [reply packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx:48]: Not sure what Nikita was after here. You can leave it as is for now.

File: packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx (hunk 2 above)
Comment (samuelmale) [reply packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx:26]: @nmalyschkin somehow my typescript compiler will complain. I think it's confusing this with being a ternary operator.

Screenshot 2021-07-19 at 10 27 49 (2)

**Comment by samuelmale** [comment]
@FlorianRappl and @nmalyschkin updated PR

File: packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx (hunk 2 above)
Comment (FlorianRappl) [reply packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx:27]: No. You have a syntax error. I think you meant `getValue?.(patient)`. What you wrote misses the alternative branch specified after an `:` (so TS is right).

Comment (samuelmale) [reply packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx]: Resolved!

**Review by FlorianRappl — APPROVED** [review]
LGTM - thanks! :beers:

**Pull request merged**